data/filter/FieldFilter.js

```diff
@@ -144,7 +146,9 @@ export class FieldFilter extends Filter {
      * @returns {boolean} - true if the other filter is fully equivalent with this instance.
      */
     equals(other) {
-        return other.isFieldFilter && isEqual(this.serialize(), other.serialize());
+        return other.isFieldFilter &&
+            isEqual(this.serialize(), other.serialize()) &&
```

Comment:
Any need to use `isEqual()` here? It won't do anything special with these two strings that `==` wouldn't do, will it? Just checking to ensure I'm not missing anything.

Reply:
nice catch